acasclient/selfile.py

```diff
@@ -0,0 +1,847 @@
+    @assay_date.setter
+    def assay_date(self, value):
+        self.meta_has_changed = (value != self._assay_date)
+        self._assay_date = value
+
+
+    @property
+    def project(self):
+        """Project (Meta Data)"""
+        return self._project
+    @project.setter
+    def project(self, value):
+        self.meta_has_changed = (value != self._project)
+        self._project = value
+
+    @property
+    def id(self):
+        """ID to uniquely identify the experiment if it is registered in the database."""
+        return self._id
+    @id.setter
+    def id(self, value):
+        self._id = value
+
+    @property
+    def expt_df(self):
+        """pd.DataFrame container for the experimental results data.  Also aliased as calculated_results_df"""
+        return self._expt_df.copy()
+    @expt_df.setter
+    def expt_df(self, new_expt_df):
+        self._expt_has_changed = self._expt_df.equals(new_expt_df)
+        self._expt_df = new_expt_df
+
+    @property
+    def calculated_results_df(self):
+        """
+        pd.DataFrame container for the 'Calculated Results'.
+        :note:  The df.columns are just the 'Corporate Batch ID' and endpoints, rows represent the values to register.  See calculated_results_datatype for the 'Datatype' mapping of the endpoints to type.
+        """
+        return self._expt_df.copy()
+
+    @calculated_results_df.setter
+    def calculated_results_df(self, new_expt_df):
+        self._expt_has_changed = self._expt_df.equals(new_expt_df)
+        self._expt_df = new_expt_df
+
+
+    @property
+    def datatype(self):
+        """OrderedDict {(column1: datatype1), ...)} column name keys for datatype."""
+        return self._datatype
+    @datatype.setter
+    def datatype(self, new_dict):
+        self._datatype_has_changed = (self._datatype == new_dict)
+        self._datatype = new_dict
+
+
+    @property
+    def calculated_results_datatype(self):
+        """OrderedDict {(column1: datatype1), ...)} column name keys for Calculated Results datatype."""
+        return self._datatype
+    @calculated_results_datatype.setter
+    def calculated_results_datatype(self, new_dict):
+        self._datatype_has_changed = (self._datatype == new_dict)
+        self._datatype = new_dict
+
+
+class Generic(AbstractExperiment):
+    """
+    Class to represent a Generic ACAS experiment file.
+    Logical units (meta data, datatype, and calculated_results) are
+    represented as separate data members.  There is some internal
+    tracking to detect if the original data content is altered, but
+    the datatype changes need to be updated by the caller.
+    calculated_results_datatype - Datatype endpoint mapping.
+    calculated_results_df - Calculated Results dataframe.
+    Common formats are supported for both input and output:  csv,
+    and excel
+    API examples:
+    # Read an existing file, change the 'Project' metadata and save.
+    import datateam.fileio.acas as acas
```

Comment:
Shawn's right about the API examples referencing datateam libraries. We should also remove this line

Reply:
Thanks, I fixed it to import from acasclient.selfile